backend/src/v1/email/email.service.ts

```diff
@@ -0,0 +1,173 @@
+import { AllegationComplaintDto } from "../../types/models/complaints/allegation-complaint";
+import { GeoOrganizationUnitCodeService } from "../../v1/geo_organization_unit_code/geo_organization_unit_code.service";
+import { AgencyCodeService } from "../../v1/agency_code/agency_code.service";
+import { ViolationCodeService } from "../../v1/violation_code/violation_code.service";
+import { GeneralIncidentComplaintDto } from "../../types/models/complaints/gir-complaint";
+import { GirTypeCodeService } from "../../v1/gir_type_code/gir_type_code.service";
+
+@Injectable()
+export class EmailService {
+  @Inject(ChesService)
+  private readonly _chesService: ChesService;
+  @Inject(DocumentService)
+  private readonly _documentService: DocumentService;
+  @Inject(EmailReferenceService)
+  private readonly _emailReferenceService: EmailReferenceService;
+  @Inject(ComplaintService)
+  private readonly _complaintService: ComplaintService;
+  @Inject(SpeciesCodeService)
+  private readonly _speciesCodeService: SpeciesCodeService;
+  @Inject(HwcrComplaintNatureCodeService)
+  private readonly _natureOfComplaintService: HwcrComplaintNatureCodeService;
+  @Inject(GeoOrganizationUnitCodeService)
+  private readonly _geoOrganizationUnitCodeService: GeoOrganizationUnitCodeService;
+  @Inject(AgencyCodeService)
+  private readonly _agencyCodeService: AgencyCodeService;
+  @Inject(ViolationCodeService)
+  private readonly _violationCodeService: ViolationCodeService;
+  @Inject(GirTypeCodeService)
+  private readonly _girTypeCodeService: GirTypeCodeService;
+  private readonly logger = new Logger(EmailService.name);
+
+  sendReferralEmail = async (createComplaintReferralDto, token, user) => {
+    try {
+      const {
+        complaint_identifier: id,
+        referred_to_agency_code,
+        referred_by_agency_code,
+        referral_reason,
+        complaint_url,
+      } = createComplaintReferralDto;
+      const { type, fileName, tz, attachments } = createComplaintReferralDto.documentExportParams;
+
+      const complaint = await this._complaintService.findById(id, type);
+
+      // Get the complaint export to be attached to the email
+      const complaintExport = await this._documentService.exportComplaint(id, type, fileName, tz, attachments, token);
+      // Convert the PDF data to base64
+      const base64Content = Buffer.from(complaintExport.data).toString("base64");
+      const emailAttachments = [
+        {
+          content: base64Content,
+          contentType: "application/pdf",
+          encoding: "base64",
+          filename: fileName,
+        },
+      ];
+
+      const senderEmailAddress = user.email ?? process.env.CEDS_EMAIL;
+      const supportEmail = process.env.CEDS_EMAIL;
+      const { given_name, family_name } = user;
+      const senderName = `${given_name} ${family_name}`;
+
+      const recipientList = [];
+      const emailReferences = await this._emailReferenceService.findActiveByAgency(referred_to_agency_code);
```

Comment:
I'm wondering if this would be a little cleaner if findActiveByAgency took an optional parameter of the geo_organization_unit_code? The switch might not be needed at all and there would be a small if (COS) pass complaint.organization.zone else null conditional instead. Not requesting this change, only a thought.

Reply:
A good one at that, I had started out that way and it's definitely a little cleaner that way at the moment. I went this route because we had discussed that area specific shared inboxes were a real possibility, and in the list of email addresses from COS there were a couple of seemingly complaint-type specific (e.g. a general_investigations_section and aquatic.invasive.species) so I tried to keep it generic

Reply:
It's not a bad thought if we're going to have future agencies enabled for referrals and added. Might it be better to just raise an exception if there's no email address configured? Something wasn't setup right, throw an error so it gets fixed asap instead of suppressing it?